src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +173,113 @@ export const handleRedirect: RequestHandler<
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicForm = form.getPublicView()
+  const { authType } = form
+  let myInfoError = false
+
+  // NOTE: Creating a variable to ensure that TS will enforce the type and ensure all keys in AuthType are covered.
+  const formController: FormController<
+    | ReturnType<typeof SpcpFactory.createFormWithSpcpSession>
+    | ReturnType<typeof MyInfoFactory.createFormWithMyInfo>
+  > = {
+    [AuthType.SP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.CP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.MyInfo]: () => {
+      return MyInfoFactory.createFormWithMyInfo(form, req.cookies)
+        .andThen((publicForm) => {
+          return extractSuccessfulMyInfoCookie(req.cookies).map(
+            (myInfoCookie) => {
+              const cookiePayload: MyInfoCookiePayload = {
+                ...myInfoCookie,
+                usedCount: myInfoCookie.usedCount + 1,
+              }
+              // NOTE: This is a side effect to set the cookie on the result after it has been successfully prefilled.
+              res.cookie(
+                MYINFO_COOKIE_NAME,
+                cookiePayload,
+                MYINFO_COOKIE_OPTIONS,
+              )
+              return publicForm
+            },
+          )
+        })
+        .mapErr((error) => {
+          // NOTE: This is a side-effect as there is no need for cookie if data could not be retrieved
+          res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
+          // NOTE: This is done as a workaround because the type of FormController enforces a uniform return value
+          // but we are required to signal if myInfoError in the event of an error
+          myInfoError = true
+          return error
+        })
+    },
+    [AuthType.NIL]: () => okAsync({ form: form.getPublicView() }),
+  }
+
+  // NOTE: Once there is a valid form retrieved from the database,
+  // the client should always get a 200 response with the form's public view.
+  // Additional errors should be tagged onto the response object like myInfoError.
+  return formController[authType]()
+    .andThen((publicFormView) =>
+      FormService.setIsIntranetFormAccess(getRequestIp(req), publicFormView),
+    )
+    .andThen((publicFormView) =>
+      FormService.setMyInfoError(publicFormView, myInfoError),
+    )
+    .map((publicFormView) => res.json(publicFormView))
+    .mapErr((error) => {
+      if (error instanceof VerifyJwtError || error instanceof InvalidJwtError) {
+        logger.error({
+          message: 'Error getting public form',
+          meta: {
+            action: 'handleGetPublicForm',
+            ...createReqMeta(req),
+            formId,
+          },
+          error,
+        })
+      }
+
+      return res.json({
+        form: publicForm,
+        myInfoError,
```

Comment:
Not sure if we need to return anything else other that the publicFormView at this point.